schema.prisma

```diff
@@ -0,0 +1,682 @@
+  Schedule     Schedule[]
+}
+
+model ClassroomTranslations {
+  language Language
+
+  /// 教室詳細資料
+  detail String
+
+  classroomId String    @db.Uuid
+  classroom   Classroom @relation(fields: [classroomId], references: [id])
+
+  @@id([language, classroomId])
+}
+
+/// 實習分組，來自舊系統 NOL table `cougrp`
+model PracticeGroup {
+  id            String @id @db.Uuid
+  // parsed with `semester`+`identifier`
+  courseId      String @db.Uuid
+  ///組別
+  group         String @db.VarChar(1)
+  /// 學生人數
+  studentNumber Int
+  /// 教學助理人數
+  taNumber      Int
+
+  // TODO: 設計課程時間關聯
+  // schedule Schedule[]
+
+  translations PracticeGroupTranslations[]
+
+  course Course @relation(fields: [courseId], references: [id])
+}
+
+model PracticeGroupTranslations {
+  language        Language
+  practiceGroupId String   @db.Uuid
+
+  comment String
+
+  practiceGroup PracticeGroup @relation(fields: [practiceGroupId], references: [id])
+
+  @@id([language, practiceGroupId])
+}
+
+// ########################################
+//  核心能力相關
+// ########################################
+
+// TODO: fix relations, translation model
+
+// model CourseCapability {
+//   /// 課程識別碼
+//   courseId String @db.VarChar(9)
+//   // 待確認是否需要
+//   // deparmentId String @db.VarChar(4)
+//   /// 建議修課年級，[1-7]
+//   recommendedGrades Int[]  
+// }
+
+/// 系所核心能力
+model DeparmentCapability {
+  /// 系所代碼
+  deparmentId    String @db.VarChar(4)
+  /// 核心代碼
+  capabilityCode String @db.VarChar(1)
+
+  @@unique([deparmentId, capabilityCode])
+}
+
+/// 核心能力類型
+model CapabilityType {
+  /// 核心代碼， [A-Z]
+  code        String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw    String @db.VarChar(100)
+  /// 英文名曾
+  nameEn      String @db.VarChar(4000)
+  /// 中文說明
+  commentZhTw String @db.VarChar(150)
+  /// 英文說明
+  commentEn   String @db.VarChar(4000)
+}
+
+/// 課程分類
+model CourseCustomType {
+  /// 課程分類代碼， [A-Z]
+  code     String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw String @db.VarChar(50)
+  /// 英文名曾
+  nameEn   String @db.VarChar(50)
+}
+
+// ########################################
+//  領域專長
+// ########################################
+
+/// 領域專長說明主表
+model Domain {
+  /// 領域專長代碼
+  code   String  @id @db.VarChar(10)
+  /// 名稱
+  name   String  @db.VarChar(50)
+  /// 英文名稱
+  nameEn String  @db.VarChar(100)
+  /// 是否通過
+  apply  Boolean
+}
+
+/// 課程領域專長關聯
+model CourseDomain {
+  /// 課程識別碼
+  CourseId   String @db.VarChar(9)
+  /// 領域專長代碼
+  DomainCode String @db.VarChar(10)
+
+  @@unique([CourseId, DomainCode])
+}
+
+// ########################################
+// 課程大綱 from ceiba db
+// ########################################
+
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+
+/// 課程大綱主表
+model Syllabus {
+  // 原sn
+  id       Int    @id
+  /// 學期，如103-1
+  semester String @db.VarChar(6)
+  /// 課程識別碼
+  // CEIBA 課程網頁有非正式課程及正式課程兩種。非正式課程指不在正式課程中的課程。
+  // 正式課程，
+  // 課程識別碼如 101 80002
+
+  // 非正式課程，
+  // 課程識別碼，前三碼為XXX 80002
+  courseNo       String                 @db.VarChar(10)
```

Comment:
Will this be related to other tables in the future? If so, we better have a better naming.

Reply:
I think the exact `courseId` should be used, which can be resolved by `semester`+`courseNo` maybe?

Reply:
Just confirm, Is that just a combination of semester and courseNo? e.g. 111 ABCDE.

If so, just create a DB function for its creation.

Reply:
Nope. `semester` with `courseNo` is a key in the model `Course`, so we can find a unique `courseId` and store it as a fkey in `Syllabus`.